### DISCUSSION TITLE
can I perform multi insert on my model?

### DISCUSSION THREAD

**Discussion by christmex** [Laravel-Backpack/community-forum#484] [Q&A (Help)]
Hi guys so I have a question, can I perform multi-insert on my current model, I will provide my ERD and the scenario detail.

here's my ERD
![image](https://github.com/Laravel-Backpack/community-forum/assets/34405837/10a452ac-f44c-4955-a12d-04ada9a3fdfe)

here's my form
![image](https://github.com/Laravel-Backpack/community-forum/assets/34405837/4bd4978f-14b8-4fd8-b0b6-49314e425e1b)

when I try to insert it, it will show an error, like this
![image](https://github.com/Laravel-Backpack/community-forum/assets/34405837/e3cd5472-bef6-4ebe-b61d-4d91cc05c70d)

here's my specific setupcreateoperation()
![image](https://github.com/Laravel-Backpack/community-forum/assets/34405837/813ba059-293c-4a33-9721-0a51bc45eb99)

little explanation, I have a transactions table that is related to a few tables, but it doesn't really matter, except for the books table, the scenario is, I want to choose more than one book from my create form, the school_year_id, semester_id,member_id they're all same and the value is not multiple just one value, so if I choose multiple books, then it will insert all the book with another data is same the difference is just the book_id, I hope you understand my question.

```
//example data
$data = [
    [
        'school_year_id' => 1,
        'semester_id' => 1,
        'member_id' => 1,
        'book_id' => 1,
    ],
    [
        'school_year_id' => 1,
        'semester_id' => 1,
        'member_id' => 1,
        'book_id' => 2,
    ],
];
```
so, like the $data, I will choose more than one book and I want to insert them at one time

should I make a new operation? or there is a way that I can use for this case? or anything? let me know, thanks for your attention

I also trying to read all of this but still, not that understand if it's the way or not
https://backpackforlaravel.com/docs/5.x/crud-how-to#hasmany-1-n-relationship
https://backpackforlaravel.com/docs/5.x/crud-fields#select_multiple-n-n-relationship

**Comment by jcastroa87** [comment]
Hi @christmex did you try add to this the "model"

`'model' => "App\Models\Book",`

Let me know if thats works for you.

Cheers.

**Reply by christmex** [reply]
Hi @jorgetwgroup , thx for the suggestion, I've tried your suggestion my code looks like this
<img width="474" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/34405837/aa98b404-6355-41a4-80de-e0f42ec2b2ad">
but the same error return
<img width="676" alt="image" src="https://github.com/Laravel-Backpack/community-forum/assets/34405837/8fbd4e06-9725-403b-b3a4-908d3e40a837">

**Comment by tabacitu** [comment]
@christmex the error message is a bit weird to me:
- first of all `book_stock` is an attribute on the `book` model... and you're in the Add Transaction form... so it shouldn't even interact with that;
- second of all, the error seems to be happening on a `collection instance`...

Where is this error happening more exactly? Are you sure this isn't some custom code that's triggering this error? What does your full error stack look like? Because Laravel is using Flare, once you get that error you should be able to "share" your entire error stack - it'll give you a link you can give us, so we have a picture of WHERE that error is happening.

Cheers!

**Reply by christmex** [reply]
Hi @tabacitu , thank you for answering, when I try to figure out why that happens I finally get the answer, that because of my validation, I've been removing the validation, the validation is to check whether the book_stock is available to loan or not, but then when I remove everything that related with validation and try to run it again, I still got this error ``` array to string error ```

here's the link from flare
https://flareapp.io/share/95JnqaB5#F75

also here's my latest controller
![image](https://github.com/Laravel-Backpack/community-forum/assets/34405837/da9b8d71-bd6b-4c5f-bda0-e71a8aa2e749)

The request body, I think nothing is wrong with this, but idk
![image](https://github.com/Laravel-Backpack/community-forum/assets/34405837/e751d57b-21ff-45be-ae78-e071de010780)


Let me know if I missing something, or maybe need other details to provide, thx for helping guys.

**Reply by tabacitu** [reply]
Oddly enough the error seems to happens after the POST request... this makes no sense. Do you have anything in your model (observers, mutators) that might use string helpers?

**Reply by pxpm** [reply]
Probably missing adding to the `$casts` property in the model ? 

```php
protected $casts = ['book_id' => 'array'];
```
What seems odd to me is that `book_**id**` is an array of *ids* and not only the book id.

**Reply by christmex** [reply]
@tabacitu Thx for replying, I don't have any mutator and when I try it, I already disabled my observer inside my model 😓

**Reply by christmex** [reply]
@pxpm Hii, thx for replying, I already tried this and still it doesn't work 🙏

**Reply by christmex** [reply]
@tabacitu @pxpm , Thanks guys for helping me, maybe I will just close this issue, if I already figure it out, I will come up with the updates, again thank you so much for helping 🙏
